### PR TITLE
fix(fetch): reject non-null window option in Request constructor

### LINKED ISSUE #5209: `Request` constructor ignores non-null `window` option instead of throwing
**Describe the bug**
`new Request()` silently accepts any value for the `window` option instead of throwing a `TypeError`. the Fetch Standard says `window` must be `null` - anything else should throw.

**To Reproduce**
```js
new Request("https://example.com", { window: 1 })
// no error in Boa, expected TypeError

new Request("https://example.com", { window: "test" })
// no error in Boa, expected TypeError
```

From repo root:
```bash
cargo run --bin boa -- -e 'try { new Request("https://example.com", { window: 1 }); console.log("OK"); } catch (e) { console.log(e.name + ": " + e.message); }'
```

Boa output:
```
OK
```

Node.js output:
```bash
node -e 'try { new Request("https://example.com", { window: 1 }); console.log("OK"); } catch (e) { console.log(e.name + ": " + e.message); }'
# TypeError: 'window' option 'client' must be null
```

**Expected behavior**
`new Request()` should throw a `TypeError` when `window` is any non-null value. `window: null` and omitting `window` should still work fine.

**Build environment:**
- OS: Windows 11
- Version: 10.0.26200
- Target triple: x86_64-pc-windows-msvc
- Rustc version: rustc 1.94.0 (4a4ef493e 2026-03-02)

**Additional context**
Root cause is in `core/runtime/src/fetch/request.rs` - `RequestInit` has no `window` field so any value passed is silently ignored.

Spec reference: https://fetch.spec.whatwg.org/#dom-request

### PR DESCRIPTION
**Pull request by HiteshShonak**
This Pull Request fixes/closes #5209.

It changes the following:
- Add `window` field to `RequestInit` and reject any non-null value with a `TypeError`.
- Add a consolidated regression test covering non-null values and confirming `null` is still accepted.

Testing:
```bash
cargo test -p boa_runtime request -- --nocapture
```

Spec reference: https://fetch.spec.whatwg.org/#dom-request

### CODE REVIEW

File: core/runtime/src/fetch/request.rs [hunk 1]
```diff
@@ -67,6 +68,12 @@ impl RequestInit {
             )?.as_str());
         }
 
+        if let Some(window) = &self.window
+            && !window.is_null()
+        {
+            return Err(js_error!(TypeError: "Request constructor: window must be null"));
+        }
```
[the remark this answers is not in the document]
Comment (HiteshShonak) [reply core/runtime/src/fetch/request.rs:75]: verified, node accepts window: undefined same as omitting it, so its current behavior is correct.

File: core/runtime/src/fetch/tests/request.rs [hunk 2]
```diff
@@ -48,6 +48,31 @@ fn request_constructor() {
...
+                    try {
+                        new Request("http://unit.test", { window: windowValue });
+                        throw Error("expected the call above to throw");
+                    } catch (e) {
+                        if (!(e instanceof TypeError)) {
+                            throw e;
+                        }
+                    }
+                }
+                new Request("http://unit.test", { window: null });
```
[the remark this answers is not in the document]
Comment (HiteshShonak) [reply core/runtime/src/fetch/tests/request.rs:70]: good point, added window: undefined assertion to the test

**Review by zhuzhu81998 — CHANGES_REQUESTED** [review]
i dont think boa is doing anything about window option at all. and this pr does not change that.

**Comment by HiteshShonak** [comment]
> i dont think boa is doing anything about window option at all. and this pr does not change that.

From the Fetch spec, window is supposed to be null, and WPT has tests that expect a TypeError when it’s non-null. Node.js also throws here.
Right now Boa just ignores the option entirely, which is what this PR is trying to fix by validating it instead of silently accepting it.
Even if some engines are lenient here, this is mainly to align with spec + WPT.

**Comment by zhuzhu81998** [comment]
> Right now Boa just ignores the option entirely, which is what this PR is trying to fix by validating it instead of silently accepting it.

 #4338 didnt implement everything from fetch.
This pr is not that meaningful, because except the trivial check, this pr is not actually implementing the missing functionality of the `window` option.

**Comment by HiteshShonak** [comment]
> This pr is not that meaningful, because except the trivial check, this pr is not actually implementing the missing functionality of the `window` option.

yeah fair, i was just trying to stop silently accepting invalid values, but fine to wait till fetch is more complete

**Comment by jedel1043** [comment]
Agreed with @zhuzhu81998, this should be part of a PR that properly implements `window` handling.
